WPMedia/arc-themes-components
# TMEDIA-709 - video player

## Ticket

- [TMEDIA-709](https://arcpublishing.atlassian.net/browse/TMEDIA-709)

## Description

Migrate video player from engine theme sdk. Used utilities pattern so that clients can create their own video player implementation if necessary. 

## Acceptance Criteria

Provide a <Video> component

Migrate the video player component from the SDK without caption object

Custom fields: 

Viewport percentage

Aspect ratio

Shrink to fit

Autoplay

Playthrough

Use all options available to Powa (docs: https://redirector.arcpublishing.com/alc/arc-products/videocenter/user-docs/video-center-video-center-player/ [Sign in | Arc Publishing](https://redirector.arcpublishing.com/alc/arc-products/videocenter/user-docs/video-center-player-settings/) )

Have Scss file that adds the hooks to the Sass library

Have test coverage

Have storybook documentation

## Test Steps

1. Checkout branch - `git checkout TMEDIA-709-video-player`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the Video and format video markup storybook documentation

## Notes

# Todo

- [x] FIgure out how aspect ratio interacts with vertical viewport width so that the div with background-color (currently hard-coded, debateable to be included) is a sufficient placeholder to minimize cls. Ideally `aspect-ratio` and `height` could be used alone with powa player -> I think `aspect-ratio` is what we want to use. `height` will need to be determined by viewport height
- [x] Set all the props available on powa, perhaps with a powa settings object, that will spread onto the powa library essentially  -> spreading the keys and values. need to link to powa docs on this and test. 
- [x] Detect video loading to remove the class "--not-loaded" upon powa initialization. Remove hard-coded isLoading state. -> removing this. as I shouldn't need to detect loading state if aspect-ratio is known
- [x] Install react-embed so that we can set the powa player code 

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
